Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp

```diff
@@ -685,14 +698,27 @@ namespace AzToolsFramework
             using namespace AZ::IO;
             AssetBrowserEntry* item = entries[0];
             bool isFolder = item->GetEntryType() == AssetBrowserEntry::AssetEntryType::Folder;
+            bool isEmptyFolder = isFolder && IsFolderEmpty(item->GetFullPath().c_str());
             Path toPath;
             Path fromPath;
             if (isFolder)
             {
-                fromPath = item->GetFullPath() + "/*";
                 Path tempPath = item->GetFullPath();
                 tempPath.ReplaceFilename(newVal.toStdString().c_str());
-                toPath = tempPath.String() + "/*";
+                // There is currently a bug in AssetProcessorBatch that doesn't handle empty folders
+                // This code is needed until that bug is fixed. GHI 13340
+                if (isEmptyFolder)
+                {
+                    fromPath = item->GetFullPath();
+                    toPath = tempPath.String();
+                    AZ::IO::SystemFile::Rename(fromPath.c_str(), toPath.c_str());
+                    return;
+                }
+                else
+                {
+                    fromPath = item->GetFullPath() + "/*";
+                    toPath = tempPath.String() + "/*";
```

Comment:
update: It may be more efficient to use `AZStd::string::format` here instead (due to the implicit string constructors that are happening behind the scenes here)